src/ngraph/pass/nop_elimination.cpp

```diff
@@ -134,62 +134,235 @@ static bool eliminate_reshape_v1(const std::shared_ptr<Node>& node)
     return false;
 }
 
+static bool replace_squeeze_unsqueeze(const std::shared_ptr<Node>& node)
+{
+    auto out_shape = node->get_output_partial_shape(0);
+    if (!out_shape.is_dynamic() && shape_size(out_shape.get_shape()) != 0)
+    {
+        shared_ptr<Node> reshape;
+        auto input = node->input_value(0).get_node_shared_ptr();
+        auto shape = node->get_shape();
+        std::vector<int64_t> vi;
+        vi.assign(shape.begin(), shape.end());
+        auto pat = op::Constant::create<int64_t>(element::i64, Shape{vi.size()}, vi);
+
+        if (as_type_ptr<opset3::Reshape>(input) || as_type_ptr<opset3::Squeeze>(input) ||
+            as_type_ptr<opset3::Unsqueeze>(input))
+        {
+            input = input->input_value(0).get_node_shared_ptr();
+            reshape = make_shared<opset3::Reshape>(input->output(0), pat, false);
```

Comment:
These are all single outputs so `output(0)` works, but it's be safer to use `node->input_value(0)` as a guard against admittedly unlikely future changes that might see a multi-output node show up there.